src/visual/style.ts

```diff
@@ -179,8 +180,9 @@ const dataColorPaletteTask: StageHandler = {
         // Each type of series use one scope.
         // Pie and funnel are using diferrent scopes
         const paletteScopeGroupByType = createHashMap<object>();
-        ecModel.eachSeries(function (seriesModel) {
-            if (!seriesModel.useColorPaletteOnData) {
+        ecModel.eachSeries((seriesModel: SeriesModel<SeriesOption & ColorByMixin>) => {
+            const colorBy = seriesModel.getColorBy();
+            if (colorBy === 'seriesId' || colorBy === 'seriesIndex' || colorBy === 'seriesName') {
```

Comment:
I think it will be much more clear if we using a method like `seriesModel.isDataLevelColorBy()` instead of checking values each time.